apps/wallet/src/ui/app/pages/home/transfer-coin/SendTokenForm.tsx

```diff
@@ -0,0 +1,324 @@
+    initialGasEstimation = 0,
+}: SendTokenFormProps) {
+    const aggregateBalances = useAppSelector(accountAggregateBalancesSelector);
+    const coinBalance = useMemo(
+        () => (coinType && aggregateBalances[coinType]) || BigInt(0),
+        [coinType, aggregateBalances]
+    );
+
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const coins = allCoins.filter(
+        ({ type }) => CoinAPI.getCoinType(type) === coinType
+    );
+
+    const suiCoins = allCoins.filter(
+        ({ type }) => CoinAPI.getCoinType(type) === SUI_TYPE_ARG
+    );
+
+    const gasAggregateBalance = aggregateBalances[SUI_TYPE_ARG] || BigInt(0);
+    const coinSymbol = (coinType && CoinAPI.getCoinSymbol(coinType)) || '';
+    const [coinDecimals, coinDecimalsQueryResult] = useCoinDecimals(coinType);
+    const [gasDecimals, gasQueryResult] = useCoinDecimals(SUI_TYPE_ARG);
+    const maxSuiSingleCoinBalance = useIndividualCoinMaxBalance(SUI_TYPE_ARG);
+
+    const validationSchemaStepOne = useMemo(
+        () =>
+            createValidationSchemaStepOne(
+                coinType || '',
+                coinBalance,
+                coinSymbol,
+                gasAggregateBalance,
+                coinDecimals,
+                gasDecimals,
+                initialGasEstimation,
+                maxSuiSingleCoinBalance
+            ),
+        [
+            coinType,
+            coinBalance,
+            coinSymbol,
+            gasAggregateBalance,
+            coinDecimals,
+            gasDecimals,
+            initialGasEstimation,
+            maxSuiSingleCoinBalance,
+        ]
+    );
+
+    const [tokenBalance, symbol, queryResult] = useFormatCoin(
+        coinBalance,
+        coinType,
+        CoinFormat.FULL
+    );
+
+    // remove the comma from the token balance
+    const formattedTokenBalance = tokenBalance.replace(/,/g, '');
+    const initAmountBig = parseAmount(initialAmount, coinDecimals);
+
+    return (
+        <Loading loading={queryResult.isLoading || gasQueryResult.isLoading || coinDecimalsQueryResult.isLoading}>
+            <Formik
+                initialValues={{
+                    amount: initialAmount,
+                    to: initialTo,
+                    isPayAllSui:
+                        !!initAmountBig &&
+                        initAmountBig === coinBalance &&
+                        coinType === SUI_TYPE_ARG,
+                    gasInputBudgetEst: initialGasEstimation,
+                }}
+                validationSchema={validationSchemaStepOne}
+                enableReinitialize
+                validateOnMount
+                validateOnChange
+                onSubmit={({
+                    to,
+                    amount,
+                    isPayAllSui,
+                    gasInputBudgetEst,
+                }: FormValues) => {
+                    if (!gasInputBudgetEst || !coins || !suiCoins) return;
+                    const coinsIDs = coins
+                        .reverse()
```

Comment:
Are we sure that coins as sorted in ascending order? I think they are stored based on object id. So I am not sure that .reverse what is expected here. We should probably use `CoinAPI.getBalance` and sort here